lib/spack/spack/spec.py

```diff
@@ -4242,7 +4268,7 @@ def from_self(name, transitive):
             else:
                 if name == other.name:
                     return False
-                if any(
+                if any(  # TODO: should this be all
```

Comment:
Does this need to be addressed in this PR?

Reply:
No, this is a future TODO. modified it to make it clearer what it's talking about for long-term tracking